fbw-a32nx/src/systems/instruments/src/MCDU/legacy/A32NX_FMCMainDisplay.ts

```diff
@@ -1321,7 +1309,7 @@ export abstract class FMCMainDisplay implements FmsDataInterface, FmsDisplayInte
     }
   }
 
-  private updatePreSelSpeedMach(preSel) {
+  private updatePreSelSpeedMach(preSel: number | undefined) {
     // The timeout is required to create a delay for the current value to be read and the new one to be set
```

Comment:
I don't really understand why this delay is there, is this still necessary or could it be removed?

Reply:
I haven't tested it without the delay. The idea is that when a new flight phase activates, we need to give the AP code time to set the preselected speed before we unset/re-set it. From the FMS, we have no control over when the FG activates the preselected speed after we fire the `K:A32NX.FMS_PRESET_SPD_ACTIVATE` event.